### PR TITLE
Only add the `galleryTpl` field to the legacy gallery element

### LINKED ISSUE #7524: Incorrect template selection in the gallery content element
### Affected version(s)

5

### Description

Im Inhaltselement Galerie stehen in den Template-Einstellungen zwei Templates zur Auswahl.
Bei Verwendung der Twig-Templates wird aber nur das Inhaltselement-Template verwendet. Das Galerie-Template wird nur in Legacy-Elementen benötigt und Anpassungen an diesem Element greifen bei Verwendung der Twig-Elemente natürlich nicht.
Bei Verwendung der Twig-Templates sollte das Galerie-Template deshalb nicht angezeigt werden.

image

### PR DESCRIPTION
**Pull request by fritzmg**
Fixes #7524

This removes the `galleryTpl` field from the palette of the `gallery` content element - but adds it again dynamically, if the registered content element is the legacy content element (or extends from it).

### CODE REVIEW

File: core-bundle/src/EventListener/DataContainer/AdjustGalleryPaletteListener.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+use Contao\ContentGallery;
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+
+/**
+ * Adjusts the palette for the "gallery" content element when the legacy version
+ * is in use.
+ */
+#[AsCallback('tl_content', 'config.onload')]
+class AdjustGalleryPaletteListener
```
Comment (m-vo) [inline core-bundle/src/EventListener/DataContainer/AdjustGalleryPaletteListener.php]: Should we maybe call it `LegacyGalleryPaletteListener`? Then it's already obvious by the filename, that this is only for the old stuff. 🙃

Comment (zoglo) [reply core-bundle/src/EventListener/DataContainer/AdjustGalleryPaletteListener.php]: Shouldn't we also add a deprecation notice to not forget to remove it in Contao 6 😃?

Comment (m-vo) [reply core-bundle/src/EventListener/DataContainer/AdjustGalleryPaletteListener.php]: The listener is not part of the public API. We can add a "todo", though. But @leofeyer will probably handle this when merging, right? ☺️

**Review by m-vo — APPROVED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**